### ISSUE TITLE
Update and testmerge pushes compiled changelogs after test merging, causing the test merge commits to get pushed too

### ISSUE THREAD

**Issue by MrStonedOne** [tgstation/tgstation-server#130]
Update and testmerge pushes compiled changelogs after test merging, causing the test merge commits to get pushed too

https://github.com/tgstation/tgstation/pull/29155

````
[00:09:50] <Mr_Bagil> REPO: Updating origin branch...(Hard Reset)
[00:09:53] <Mr_Bagil> REPO: Test merging PR #29155...
[00:09:54] <TheGhostOfWhibyl1> https://github.com/tgstation/tgstation/pull/29155 - '[wip][dnm]A* improvements' by MrStonedOne
[00:10:04] <TheGhostOfWhibyl1> [tgstation] tgstation-server pushed 6 new commits to master: https://github.com/tgstation/tgstation/compare/5836c10e3097...86c41964b99f
[00:10:04] <TheGhostOfWhibyl1> tgstation/master 85de154 MrStonedOne: astar improvements...
[00:10:04] <TheGhostOfWhibyl1> tgstation/master e5d3cd8 MrStonedOne: commiting debug version just so its on record somewhere if i want it again.
[00:10:04] <TheGhostOfWhibyl1> tgstation/master f952583 MrStonedOne: And the comparison debug version
[00:10:11] <TheGhostOfWhibyl1> [tgstation] tgstation-server closed pull request #29155: [wip][dnm]A* improvements (master...betterastar) https://github.com/tgstation/tgstation/pull/29155
[00:10:13] <TheGhostOfWhibyl1> [tgstation] tgstation-server pushed 1 new commit to master: https://github.com/tgstation/tgstation/commit/73f60400e86483abb1ab0842d30852bda5afcae0
[00:10:13] <TheGhostOfWhibyl1> tgstation/master 73f6040 tgstation-server: Automatic changelog generation for PR #29155 [ci skip]
````

@Cyberboss This can't happen. Me even having commit access to push compiled changelogs is on shaky grounds, and I kept it because I made damn sure shit like this never happened by treating the compile and push changelog code in tgs2 with more care and attention then I treat player facing non-parameterized sql code. I basically treated it like a loaded bomb.

**Comment by JamieH** [comment]
One solution to this whole thing is you run the repo on a "server" branch, and then you compile the changelog on the server branch, copy the files to master branch and make a commit or cherry pick your commit from the server branch to the master branch.

**Comment by RemieRichards** [comment]
@JamieH that's a workaround, not a fix.

**Comment by JamieH** [comment]
@RemieRichards How would you suggest fixing it then?

**Comment by RemieRichards** [comment]
The Pull request I have open that's also linked on this issue?

**Comment by JamieH** [comment]
That PR won't work, there are still edge cases where the changelog would be pushed.

**Comment by RemieRichards** [comment]
Those 'edge cases' are Cyberboss' error.
I fixed the one that specifically tripped up MSO, please, do share the others so we can fix them.

Using a workaround forever is not acceptable.

**Comment by JamieH** [comment]
Apart from the fact that it's not a workaround and it's the ideal use for git branches. What if there is a line of code someone doesn't want pushed to their repo? 

Actually your change doesn't work at all - since you can have testmerges already merged before the `UpdateServer` function runs. The only case it ~might~ work is if you did a hard update and test merged one PR at the same time.

**Comment by RemieRichards** [comment]
Whatever.

**Comment by optimumtact** [comment]
yall pump the brakes yo, it's two different ways of doing things

**Comment by optimumtact** [comment]
was this fixed?

**Comment by Cyberboss** [comment]
Nope

**Comment by Cyberboss** [comment]
Fixed in #134
